server/webapp/WEB-INF/rails.new/app/helpers/api_v1/authentication_helper.rb

```diff
@@ -74,5 +83,15 @@ def render_unauthorized_error
       render :json_hal_v1 => { :message => 'You are not authorized to perform this action.' }, :status => 401
     end
 
+    private
+    def is_user_a_group_admin?(current_user, group_name)
```

Comment:
Most of this logic including the check for `security_service.isSecurityEnabled` is already taken care of as part of [SecurityService.isUserAdminOfGroup](https://github.com/gocd/gocd/blob/824f940a5163d9c90e6e34747f87baf3e6f7f84c/server/src/com/thoughtworks/go/server/service/SecurityService.java#L69). 

`goConfigService.groups().hasGroup(groupName) && !securityService.isUserAdminOfGroup(currentUser.getUsername(), groupName)` should be good enough check